### ISSUE TITLE
ImportError: No module named spi_serial with 0.5.x and Non Explorer board

### ISSUE THREAD

**Issue by PieterGit** [openaps/oref0#552]
@LilDucky reports and error with Carelink USB, RPi2 and MDT CGM
https://gitter.im/nightscout/intend-to-bolus?at=596d4fc81c8697534a342b07

```
MDT CGM configured; not waiting
ImportError: No module named spi_serial
```

I think this error will exist for all non explorer board (pump device not `/dev/spidev5.1`) users with 0.5.x.
Note that the error is harmless, and the pump init and communication should just work fine. But we should avoid false error messages if we can.

It's caused by this pull request of https://github.com/openaps/oref0/pull/498 of @Kdisimone 

I noticed it, but didn't come by to fix it properly. I think the follow options can solve this bug:

Option 1. short term: don't add `reset_spi_serial.py 2>/dev/null;` to `lib/oref0-setup/alias.json` for users that don't have pump device `/dev/spidev5.1`

Option 2. fix improve oref0_init_pump_comms.py so that it will always invoke the reset_spi_serial.py.
The code is there, but somehow   @Kdisimone  and @scottleibrand had problems that it didn't executed the reset_spi_serial in some/all cases. The `oref0_init_pump_comms.py` has the code and logic to invoke the reset_spi_serial, here: https://github.com/openaps/oref0/blob/dev/bin/oref0_init_pump_comms.py#L23-L38

Perhaps not checking for the import, but checking for the pump device is better and will fix so that the extra `reset_spi_serial.py 2>/dev/null;` can be removed.

Option 3. (that's where I was heading for). Rewrite `oref0_init_pump_comms.py` to a `openaps use pump init` command that will call reset_spi_serial.py logic and init WW-pump (if applicable). That code should also be called by openaps if you use `openaps mmtune`. I tried to get that in, but my Python / openaps skills did not succeed yet.

One possible problem with option 2 is that there is some Python2 / Python3 thing. I wrote `oref0_init_pump_comms.py` in Python 3 because Python 3 has far better options of killing processes after a timeout. Killing a hanging pump WW init can also be handled by the `killall -g openaps ` in the crontab, but that takes much longer.

Option 2 and 3 also get rid of the double reset_spi_serial.py call, that has no purpose.
I prefer option 3 is that is, because it's the cleanest python and most openaps way 
But it will require a new openaps release and will force users to upgrade their openaps version if they want to use oref0 and  `oref0-setup.sh`.

Thoughts?

**Comment by PieterGit** [comment]
@LilDucky : can you tell me on which type of rig you experienced this? RPI3/Explorerboard.
Can you confirm you use Carelink for communication to your pump (and CGM)

**Comment by LilDucky** [comment]
@PieterGit I am on an rPi2 with carelink stick.

**Comment by PieterGit** [comment]
workaround for you @LilDucky is to remove the `reset_spi_serial.py 2>/dev/null; ` in the mmtune alias in `openaps.ini` file. That will reverse https://github.com/openaps/oref0/pull/498/files for you. Can you please respond if that fixes all the  `ImportError: No module named spi_serial`. 

Please also add a `-v` to your `oref0_init_pump_comms.py`  because that will tell a bit more of what's happening. So the mmtune alias should start with `oref0_init_pump_comms.py -v ` followed by the `find ...` stuff.

**Comment by Kdisimone** [comment]
Original change in the alias was due to issue #522 being encountered by many.

**Comment by PieterGit** [comment]
@LilDucky error is probably caused by theses lines:
https://github.com/openaps/oref0/blob/master/bin/oref0-pump-loop.sh#L319-L320

Can you comment `reset_spi_serial.py 2>/dev/null`  out (change it to `#reset_spi_serial.py 2>/dev/null`

**Comment by PieterGit** [comment]
I think the fix for @LilDucky error will be need to be applied to https://github.com/openaps/oref0/blob/master/bin/oref0-pump-loop.sh#L319-L320

`reset_spi_serial.py`  must be conditinally run if `/dev/spidev5.1` is used as pump device
I'll see if `oref0_init_pump_comms.py` can be changed so that it won't show the error.

Asking @scottleibrand about his opinion/advice.

The error use case I described is still there, but not with RPI[23]/Carelink, but probably applies to RPI[23]/TI USB. I'll test that later.

**Comment by PieterGit** [comment]
@LilDucky : you reported that it still doesn't work. I suspect it's an install thing.
Can you check your `/usr/local/bin/oref0-pump-loop`.
Line 317 and further should now be:
``` 
function mmtune {
    # TODO: remove reset_spi_serial.py once oref0_init_pump_comms.py is fixed to do it correctly
    #reset_spi_serial.py 2>/dev/null
    #oref0_init_pump_comms.py
```

**Comment by LilDucky** [comment]
only the reset_spi line is commented out.
rebooted and running 
`cd ~/src/oref0 ; npm run global-install`
/usr/local/bin/oref0-pump-loop contains
`function mmtune {
    # TODO: remove reset_spi_serial.py once oref0_init_pump_comms.py is fixed to do it correctly
#    reset_spi_serial.py 2>/dev/null
    oref0_init_pump_comms.py
    echo -n "Listening for 30s silence before mmtuning: "
    for i in $(seq 1 800); do
        echo -n .
        mmeowlink-any-pump-comms.py --port $port --wait-for 30 2>/dev/null | egrep -v subg | egrep No \
        && break
    done
    echo {} > monitor/mmtune.json
    echo -n "mmtune: " && openaps report invoke monitor/mmtune.json 2>&1 >/dev/null | tail -1
    grep -v setFreq monitor/mmtune.json | grep -A2 $(json -a setFreq -f monitor/mmtune.json) | while read line
        do echo -n "$line "
    done
    rssi_wait=$(grep -v setFreq monitor/mmtune.json | grep -A2 $(json -a setFreq -f monitor/mmtune.json) | tail -1 | awk '($1 < -60) {print -($1+60)*2}')
    if [[ $rssi_wait > 1 ]]; then
        echo "waiting for $rssi_wait second silence before continuing"
        wait_for_silence $rssi_wait
    fi
}

`
I will comment out the first 2 lines and see what happens now

**Comment by LilDucky** [comment]
now getting
`Starting pump-loop at Wed 19 Jul 16:49:05 AEST 2017:
MDT CGM configured; not waiting
ImportError: No module named spi_serial
Listening for 30s silence before mmtuning: .......................`

I think the entire routine needs culling for me.

**Comment by LilDucky** [comment]
same result running from command line

/usr/local/bin/oref0-pump-loop contains:
```
function mmtune {
    # TODO: remove reset_spi_serial.py once oref0_init_pump_comms.py is fixed to do it correctly
#    reset_spi_serial.py 2>/dev/null
#    oref0_init_pump_comms.py
    echo -n "Listening for 30s silence before mmtuning: "
    for i in $(seq 1 800); do
       echo -n .
       mmeowlink-any-pump-comms.py --port $port --wait-for 30 2>/dev/null | egrep -v subg | egrep No \
        && break
    done
```

Output from command line is
```
pi@raspberrypi:~/lilducky $ oref0-pump-loop

Starting pump-loop at Wed 19 Jul 17:48:16 AEST 2017:
MDT CGM configured; not waiting
ImportError: No module named spi_serial
Listening for 30s silence before mmtuning: .....^C
```

**Comment by PieterGit** [comment]
I don't know where the the error message occurs from. It must be from a `import spi_serial`, but apparently it's not in the mmtune alias or in the mmtune function of the `oref0-pump-loop`. Help wanted. I'll hope to work on rewriting the pump init (see option 3 in the issue description), but that won't be in the next weeks.

**Comment by scottleibrand** [comment]
Added the check to only reset_spi_serial if port is /dev/spidev5.1 in https://github.com/openaps/oref0/pull/569 if you'd like to test that...

**Comment by scottleibrand** [comment]
@LilDucky If you're getting a spi_serial import error without an mmtune or explicit reset_spi_serial call, perhaps the mmeowlink-any-pump-comms.py in wait_for_silence is eventually calling this code? 
https://github.com/oskarpearson/mmeowlink/blob/dev/mmeowlink/vendors/serial_interface.py#L25-L27
I think this will only trigger an import attempt if a SPI device actually exists on your system.  What does `ls -la /dev/*spi*` show on that rig?

**Comment by LilDucky** [comment]
@scottleibrand I am away from home but will look this afternoon (in about 5 hours)

**Comment by LilDucky** [comment]
@scottleibrand there are no device entries in /dev with spi in the name.

```
root@raspberrypi:/dev# ls -la /dev/*spi*
ls: cannot access /dev/*spi*: No such file or directory
```

**Comment by scottleibrand** [comment]
I'm assuming this is now fixed with #569 - please reopen if you still see issues with 0.6.0-dev.
